geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala

```diff
@@ -227,7 +229,7 @@ object KryoLazyDensityIterator extends LazyLogging {
    * Decodes a result feature into an iterator of (x, y, weight)
    */
   private def decodeResult(gridSnap: GridSnap)(sf: SimpleFeature): Iterator[(Double, Double, Double)] = {
-    val result = sf.getAttribute(0).asInstanceOf[Array[Byte]]
+    val result = sf.getUserData.get(DENSITY_VALUE).asInstanceOf[Array[Byte]]
```

Comment:
maybe we should add a check for the attribute as a fallback to preserve version compatibility